src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,96 @@
+package duke.util;
+
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+public class Parser {
+
+
+    /**
+     * Trims the input String into the default reading format.
+     *
+     * @param input The input String.
+     * @return The trimmed input String to be processed.
+     */
+    public static String parseInput(String input) {
+        return input.trim().replaceAll(" +", " ");
+    }
+
+    /**
+     * Parses the given String into a Command.
+     *
+     * @param input The input String.
+     * @return The specific Command to execute.
+     */
+    public static Command parseCommand(String input) {
+
+        String[] fullCommand = input.split(" ", 2);
+        Command command = null;
+        switch (fullCommand[0]) {
+        case "bye":
+            if (fullCommand.length > 1) {
+                command = new WrongCommand("You cannot put anything behind the keyword \"bye\".");
+            } else {
+                command = new ExitCommand();
+            }
+            break;
+        case "list":
+            if (fullCommand.length > 1) {
+                command = new WrongCommand("You cannot put anything behind the keyword \"list\".");
+            } else {
+                command = new ListCommand();
+            }
+            break;
+        case "todo":
+            try {
+                command = new AddCommand(ToDo.of(fullCommand[1]));
+            } catch (ArrayIndexOutOfBoundsException e) {
+                command = new WrongCommand("The description of a todo cannot be empty, idiot.");
+            }
+            break;
+        case "deadline":
+            try {
```

Comment:
As explained in deadline task, the error on improper date time formatting will cause the program to stop. Catch this error to return something useful for the user to work with.